plugin-server/src/main/ingestion-queues/session-recording-v2/retention/retention-service.ts

```diff
@@ -0,0 +1,111 @@
+import { RedisPool, Team } from '../../../../types'
+import { TeamId } from '../../../../types'
+import { BackgroundRefresher } from '../../../../utils/background-refresher'
+import { PostgresRouter, PostgresUse } from '../../../../utils/db/postgres'
+import { logger } from '../../../../utils/logger'
+import { ValidRetentionPeriods } from '../constants'
+import { MessageWithTeam } from '../teams/types'
+import { RetentionPeriod } from '../types'
+import { MessageWithRetention } from './types'
+
+function isValidRetentionPeriod(retentionPeriod: string): retentionPeriod is RetentionPeriod {
+    return ValidRetentionPeriods.includes(retentionPeriod as RetentionPeriod)
+}
+
+export class RetentionService {
+    private readonly retentionRefresher: BackgroundRefresher<Record<TeamId, RetentionPeriod>>
+
+    constructor(
+        private postgres: PostgresRouter,
+        private redisPool: RedisPool,
+        private keyPrefix = '@posthog/replay/'
+    ) {
+        this.retentionRefresher = new BackgroundRefresher(
+            () => this.fetchTeamRetentionPeriods(),
+            5 * 60 * 1000, // 5 minutes
+            (e) => {
+                // We ignore the error and wait for postgres to recover
+                logger.error('Error refreshing team retention periods', e)
```

Comment:
we could have a prom counter here 🤔 trying to picture what grafana dashboard I'd want to make to know if this all was working when deployed (and what we might want to alert on...)

Reply:
Will add an error counter 👍